### PR TITLE
Added CPK-N

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs [hunk 1]
```diff
@@ -81,6 +82,16 @@ public class BlobContainerClient
         /// </summary>
         internal virtual CustomerProvidedKey? CustomerProvidedKey => _customerProvidedKey;
 
+        /// <summary>
+        /// The <see cref="EncryptionScope"/> to be used when sending requests.
+        /// </summary>
+        internal readonly EncryptionScope _encryptionScope;
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: I'm leaning toward removing this property from BlobContainer, it isn't actually used on any operations, and I think users may find the ambiguity between the BlobContainerClient.Create() ContainerEncryptionScopeOptions parameter and this property confusing.

Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: > I'm leaning toward removing this property from BlobContainer, it isn't actually used on any operations, and I think users may find the ambiguity between the BlobContainerClient.Create() ContainerEncryptionScopeOptions parameter and this property confusing.

The issue I see with this is that the BlobClientOptions property would have no effect at all for Containers.

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/Models/BlobsModelFactory.cs [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+
+namespace Azure.Storage.Blobs.Models
+{
+    public partial class BlobsModelFactory
```
[the remark this answers is not in the document]
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Models/BlobsModelFactory.cs]: added.

File: sdk/storage/Azure.Storage.Blobs/src/Models/ContainerEncryptionScopeOptions.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        /// <summary>
+        /// Specifies the default encryption scope to set on the container and use for all future writes.
+        /// </summary>
+        public string DefaultEncryptionScope { get; set; }
+
+        /// <summary>
+        /// If true, prevents any request from specifying a different encryption scope than the scope set on the container.
+        /// </summary>
+        public bool DencyEncryptionScopeOverride { get; set; }
```
[the remark this answers is not in the document]
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Models/ContainerEncryptionScopeOptions.cs]: fixed.

File: sdk/storage/Azure.Storage.Blobs/src/Models/EncryptionScope.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Azure.Storage.Blobs.Models
+{
+    /// <summary>
+    /// Wrapper for an encryption scope for server-side encryption.
+    /// </summary>
+    public class EncryptionScope
+    {
+        /// <summary>
+        /// The name of a previously-createdencryption scope created in the control plane, to be used for server-side encryption.
```
[the remark this answers is not in the document]
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Models/EncryptionScope.cs]: fixed.

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**